lib/galaxy/model/__init__.py

```diff
@@ -846,7 +846,7 @@ def get_param_values( self, app ):
         dict of tool parameter values.
         """
         param_dict = dict( [ ( p.name, p.value ) for p in self.parent_job.parameters ] )
-        tool = app.toolbox.get_tool( self.tool_id )
+        tool = app.toolbox.get_tool( self.tool_id, tool_version=self.job.tool_version )
```

Comment:
I think that `self.tool_id` should be changed to `self.job.tool_id` . I suspect this method may be unused.